docs/meson.build

```diff
@@ -43,6 +43,17 @@ if gidocgen_dep.found() and docs_python_deps.allowed() and gidocgen_app.found()
     install_dir: join_paths(datadir, 'doc'),
   )
 
+  hsi_md = custom_target('generate-hsi-spec',
+    input : 'hsi.md',
+    output : 'hsi.md',
+    command : [
+      join_paths(meson.current_source_dir(), 'generate-hsi-spec.py'),
```

Comment:
I think you shouldn't assume it's executable and use the default interpreter from the shebang.  You should use python as found by meson to execute it

Reply:
Got it, thanks. Fixed.

Reply:
Truthfully, since this code doesn't seem to care what version of python it runs using, you could specify the shebang as `/usr/bin/env python3`.

Random aside: "use python as found by meson" -- if you ***remove*** the executable bit from the script, Meson notices that it cannot be run as a command, so it reads the shebang itself and runs the interpreter with the script as an argument. For scripts with a shebang ending in python3, it then always uses Meson's own internal python (might be /usr/bin/python3, might be `meson.exe runpython` on Windows).